src/Fmt.ml

```diff
@@ -29,6 +29,16 @@ let break n o fs = Format.pp_print_break fs n o
 
 let fmt f fs = Format.fprintf fs f
 
+let eol_str () =
+  match !Conf.current_line_endings with `Unix -> "\n" | `Windows -> "\r\n"
+
+let eol_fmt () =
+  match !Conf.current_line_endings with
+  | `Unix -> Caml.format_of_string "\n"
+  | `Windows -> Caml.format_of_string "\r\n"
+
+let double_eol_fmt () = Caml.( ^^ ) (eol_fmt ()) "@\n"
```

Comment:
I'm wary of dynamic format string construction as I have had performance problems with it in the past.